packages/minter-contracts/test-hs/Test/Marketplace/FA2FixedFee.hs

```diff
@@ -0,0 +1,377 @@
+module Test.Marketplace.FA2FixedFee where
+
+import qualified Data.Map as Map
+import Hedgehog (Gen, Property, forAll, property)
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import Morley.Nettest
+
+import Lorentz hiding (balance, contract)
+import Lorentz.Test (contractConsumer)
+import Michelson.Typed (convertContract, untypeValue)
+
+import qualified Indigo.Contracts.FA2Sample as FA2
+import Lorentz.Contracts.Marketplace.FA2FixedFee
+import Lorentz.Contracts.PausableAdminOption
+import Lorentz.Contracts.Spec.FA2Interface (TokenId(TokenId))
+import qualified Lorentz.Contracts.Spec.FA2Interface as FA2
+import Test.Util
+
+hprop_Every_sale_sends_a_fee_to_the_fee_collector :: Property
+hprop_Every_sale_sends_a_fee_to_the_fee_collector =
+  property $ do
+    testData@TestData{testSalePrice, testFeePercent} <- forAll genTestData
+    clevelandProp $ do
+      setup@Setup{seller, buyer, feeCollector, moneyFA2} <- testSetup testData
+      contract <- originateMarketplaceContract setup
+      consumer <- originateSimple "balance-response-consumer" [] (contractConsumer @[FA2.BalanceResponseItem])
```

Comment:
This is repeated several time, it could be extracted to a function.

Not sure if it makes sense to include it in the `balanceOf` itself for simplicity, as I see there are several tests where that's called more than once.

Reply:
Yeah, at first I thought it would be best _not_ to put it in `balanceOf`, for performance reasons.

But seeing as these tests are only run in the emulator anyway, then the cost of creating multiple consumers per test is negligible. I've tried this and can confirm it doesn't affect the test suite's run time.

I've made this change.